src/Console/Migration/UnsignedKeysCommand.php

```diff
@@ -105,27 +126,60 @@ protected function execute(InputInterface $input, OutputInterface $output)
-                    ) {
+                // Check for negative values in table data
+                $min = $this->db
+                    ->request(['SELECT' => ['MIN' => sprintf('%s AS min', $column_name)], 'FROM' => $table_name])
+                    ->current()['min'];
+                if ($min !== null && $min < 0) {
+                    if (!preg_match('/^glpi_plugin_/', $table_name)) {
+                        // Force migration of unconsistent -1 values in core tables
+                        $forced_value = $default !== null || $nullable ? $default : 0;
                         $message = sprintf(
-                            __('Migration of column "%s.%s" cannot be done as it is referenced in CONSTRAINT "%s" of table "%s.%s".'),
+                            __('Column "%s.%s" contains negative values. Updating them to "%s"...'),
                             $table_name,
                             $column_name,
-                            $foreign_key['CONSTRAINT_NAME'],
-                            $foreign_key['TABLE_NAME'],
-                            $foreign_key['COLUMN_NAME']
+                            $forced_value === null ? 'NULL' : $forced_value
+                        );
+                        $this->writelnOutputWithProgressBar(
+                            '<comment>' . $message . '</comment>',
+                            $progress_bar
+                        );
+                        $result = $this->db->update(
+                            $table_name,
+                            [$column_name => $forced_value],
+                            [$column_name => ['<', 0]]
+                        );
+                        if ($result === false) {
+                            $message = sprintf(
+                                __('Updating column "%s.%s" values failed with message "(%s) %s".'),
+                                $table_name,
+                                $column_name,
+                                $this->db->errno(),
+                                $this->db->error()
+                            );
+                            $this->writelnOutputWithProgressBar(
+                                '<error>' . $message . '</error>',
+                                $progress_bar,
+                                OutputInterface::VERBOSITY_QUIET
+                            );
+                            $errors = true;
+                            continue; // Go to next column
+                        }
+                    } else {
+                        // Cannot determine whether -1 values in plugin tables are legitimate (bad foreign key design)
+                        // or inconsistent (wrong value inserted in DB)
+                        $message = sprintf(
+                            __('Migration of column "%s.%s" cannot be done as it contains negative values.'),
```

Comment:
Sometimes, running this migration command will be done juste after the update process, prior to plugins updates (see #11566 ). Adding following message may help to prevent having this kind of issues, that can be handled only by plugins updates.

```suggestion
                            __('Migration of column "%s.%s" cannot be done as it contains negative values. You should try to update corresponding plugin to its latest version and run the command again.'),
```

@orthagh @trasher What do you think ?

Reply:
yes i agree, i got the questions a lot and answer to check plugins update

Reply:
![image](https://user-images.githubusercontent.com/33253653/168606021-0fe8b305-f771-4c7c-8160-7b89cde6a732.png)